lti_consumer/models.py

```diff
@@ -258,6 +273,13 @@ def clean(self):
         if consumer is None:
             raise ValidationError(_("Invalid LTI configuration."))
 
+    def save(self, *args, **kwargs):
+        # Only allow creation with a CCX location if CCX master configuration exists.
+        if not self.pk and is_ccx_location(self.location) and not self.ccx_master_configuration:
```

Comment:
You are already checking is_ccx_location inside ccx_master_configuration, so, do you think you need to check it here? @kuipumu

Reply:
@Squirrel18 This is a different check, there is a chance for example, that when I'm loading a CCX and a new set of LtiConfigurations are created, and for some reason the master course LtiConfigurations where deleted, a new set of keys will be created for the CCX blocks without the author or user knowing it, this avoids creating a new LtiConfiguration on a CCX with unknown keys. It would be better to show this message so the user knows it, but I don't know how to do it with an XBlock, or disable the deletion of LtiConfigurations from the admin view to avoid chances of this scenario occurring, what do you think about this?